src/lib.rs

```diff
@@ -106,6 +113,26 @@ impl Encoder<ClamdRequestMessage> for ClamdZeroDelimitedCodec {
                 dst.put_u32(0);
                 Ok(())
             }
+            ClamdRequestMessage::StartSession => {
+                dst.reserve(10);
+                dst.put(&b"zIDSESSION"[..]);
+                dst.put_u8(0);
+                Ok(())
+            }
+            ClamdRequestMessage::EndSession => {
+                dst.reserve(10);
```

Comment:
Nitpick: that should be `dst.reserve(5)` (its the size of the command + the null byte).

Reply:
Ah yes indeed !